Mlem/App/Views/Shared/ExpandedPost/ExpandedPostView.swift

```diff
@@ -20,111 +19,114 @@ struct ExpandedPostView: View {
             value.merge(nextValue()) { $1 }
         }
     }
-    
+
     @Environment(Palette.self) var palette
     @Environment(AppState.self) var appState
     @Environment(\.dismiss) var dismiss
     
     @Setting(\.jumpButton) var jumpButton
     
     let post: AnyPost
-    @State var showCommentWithActorId: URL?
+    @State var tracker: CommentTreeTracker?
+    @State var highlightedComment: (any CommentStubProviding)?
+    @State var scrolledToHighlightedComment: Bool
+    
     @State var jumpButtonTarget: URL?
     @State var topVisibleItem: URL?
     
-    @State var tracker: ExpandedPostTracker?
+    init(post: AnyPost, highlightedComment: (any CommentStubProviding)?) {
+        self.post = post
+        self._highlightedComment = .init(wrappedValue: highlightedComment)
+        self._scrolledToHighlightedComment = .init(wrappedValue: highlightedComment == nil)
+    }
     
     var body: some View {
         ContentLoader(model: post) { proxy in
-            if let post = proxy.entity, let tracker {
-                let showLoadingSymbol = showCommentWithActorId == nil || (self.post.isUpgraded && tracker.loadingState != .loading)
+            // Using a `ZStack` here rather than `if`/`else` because there needs to
+            // be a delay between the `content()` appearing and calling `scrollTo`
```

Comment:
This can be achieved using `.overlay`:

```
            Group {
                if let post = proxy.entity, let tracker {
                    content(post: post, tracker: tracker, isLoading: proxy.isLoading)
                        .externalApiWarning(entity: post, isLoading: proxy.isLoading)
                        .task {
                            if post.api == appState.firstApi, tracker.loadingState == .idle {
                                post.markRead()
                                await load(tracker: tracker)
                            }
                        }
                } else {
                    ProgressView()
                }
            }
            .overlay {
                if showLoadingSymbol {
                    palette.background
                        .ignoresSafeArea()
                        .overlay {
                            ProgressView()
                                .tint(.secondary)
                        }
                        .transition(.opacity)
                }
            }
            .animation(.easeOut(duration: 0.1), value: showLoadingSymbol)
```